src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -288,20 +300,199 @@ private static ProjectItemGroupElement CreateItemGroup(Project project, string f
             }
         }
 
+        /// <summary>
+        /// A simple check for some of the evaluated properties to check
+        /// if the project is package reference project or not
+        /// </summary>
+        /// <param name="project"></param>
+        /// <returns></returns>
+        internal static bool IsPackageReferenceProject(Project project)
+        {
+            return (project.GetPropertyValue(RESTORE_STYLE_TAG) != "" ||
+                    project.GetPropertyValue(PACKAGE_REFERENCE_TYPE_TAG) != "" ||
+                    project.GetPropertyValue(NUGET_STYLE_TAG) != "" ||
+                    project.GetPropertyValue(ASSETS_FILE_PATH_TAG) != "");
+        }
+
+        /// <summary>
+        /// Prepares the dictionary that maps frameworks to packages top-level
+        /// and transitive.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <param name="userInputFrameworks">A list of framework names</param>
+        /// <param name="assetsFile">Assets file for all targets and libraries</param>
+        /// <param name="transitive">Include transitive packages in the result</param>
+        /// <returns></returns>
+        internal IEnumerable<FrameworkPackages> GetResolvedVersions(
+            Project project, IEnumerable<string> userInputFrameworks,LockFile assetsFile, bool transitive)
+        {
+            var resultPackages = new List<FrameworkPackages>();
+            var parsedUserFrameworks = userInputFrameworks.Select(f => NuGetFramework.Parse(f));
+
+            IEnumerable<TargetFrameworkInformation> requestedTargetFrameworks;
+
+            if (!parsedUserFrameworks.Any())
+            {
+                requestedTargetFrameworks = assetsFile.PackageSpec.TargetFrameworks;
+            }
+            else
+            {
+                requestedTargetFrameworks = assetsFile.PackageSpec.TargetFrameworks.Where(tfm => parsedUserFrameworks.Contains(tfm.FrameworkName));
+            }
+
+            var requestedTargets = requestedTargetFrameworks.Select(tfm => assetsFile.GetTarget(tfm.FrameworkName, null));
```

Comment:
As I mentioned somewhere above, you'd need to handle RID along with TFMs... @rohit21agrawal can give you more context about RIDs (RuntimeIdentifiers)

Reply:
We are not taking RID into account for now